lib/main.rb

```diff
@@ -0,0 +1,82 @@
+module Enumerable
+  def my_each
+    element = 0
+    while element < length
+      yield(self[element])
+      element += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    element = 0
+    while element < length
+      yield(self[element], element)
+      element += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    my_each do |a|
+      yield(a) ? new_array.push(a) : new_array
+    end
+  end
+
+  def my_all?
+    my_each do |a|
+      return false unless yield(a)
+    end
+    true
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ]  my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
  ```ruby
  array = [1,2,5,4,7]
  p array.my_all?(Integer) == array.all?(Integer) # false
- [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
  ```ruby
  array = [1,2,5,4,7]
  p array.my_all? == array.all? # false
- [ ] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
  ```ruby
  array = %w[dog door rod blade]
  p array.my_all?(/c/) == array.all?(/c/) # false
- [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
  ```ruby
  array = %w[dog door rod blade]
  p array.my_all?(5) == array.all?(5) # false